deployer/src/deployer/search/__init__.py

```diff
@@ -0,0 +1,191 @@
+import json
+import re
+import time
+from pathlib import Path
+from collections import defaultdict
+
+import click
+from elasticsearch.helpers import streaming_bulk
+from elasticsearch_dsl.connections import connections
+from selectolax.parser import HTMLParser
+
+from .models import Document
+
+
+def index(
+    buildroot: Path,
+    url: str,
+    update=False,
+    no_progressbar=False,
+    priority_prefixes: (str) = (),
+):
+    # We can confidently use a single host here because we're not searching
+    # a cluster.
+    connections.create_connection(hosts=[url])
+    connection = connections.get_connection()
+    health = connection.cluster.health()
+    status = health["status"]
+    if status not in ("green", "yellow"):
+        raise click.ClickException(f"status {status} not green or yellow")
+
+    count_todo = 0
+    for file in walk(buildroot):
+        count_todo += 1
+
+    click.echo(f"Found {count_todo:,} (potential) documents to index")
+
+    # Confusingly, `._index` is actually not a private API.
+    # It's the documented way you're supposed to reach it.
+    index = Document._index
+    if not update:
+        click.echo(
+            "Deleting any possible existing index "
+            f"and creating a new one called {index._name}"
+        )
+        index.delete(ignore=404)
+        index.create()
+
+    search_prefixes = [None]
+    for prefix in priority_prefixes[::-1]:
```

Comment:
I prefer `reversed(priority_prefixes)` mainly because I find it more readable, but also because it's more efficient in that it doesn't make a copy (which doesn't matter in this case, of course).